platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -213,32 +201,39 @@ export function ViewportGridProvider({ children, service }) {
         // haven't been viewed yet, and add them in the appropriate order.
         const options = {};
 
-        let activeViewportIndexToSet = activeViewportIndex;
+        let activeViewportIdToSet = activeViewportId;
         for (let row = 0; row < numRows; row++) {
           for (let col = 0; col < numCols; col++) {
             const pos = col + row * numCols;
             const layoutOption = layoutOptions[pos];
             const positionId = layoutOption?.positionId || `${col}-${row}`;
+
             if (hasOptions && pos >= layoutOptions.length) {
               continue;
             }
-            if (
-              activeViewportIndexToSet == null &&
-              state.viewports[state.activeViewportIndex]?.positionId ===
-                positionId
-            ) {
-              activeViewportIndexToSet = pos;
-            }
+
             const viewport = findOrCreateViewport(pos, positionId, options);
             if (!viewport) {
               continue;
             }
+
             viewport.positionId = positionId;
+
+            // If the viewport doesn't have a viewportId, we create one
+            if (!viewport.viewportOptions?.viewportId) {
+              const randomUID = utils.uuidv4().substring(0, 8);
+              viewport.viewportOptions.viewportId = `viewport-${randomUID}`;
+            }
+
+            if (!viewport.viewportId) {
```

Comment:
Does it make sense to simply not do this check (i.e. remove the if) and just always set `viewport.viewportId` to safeguard against (potentially) having a different viewportId in viewportOptions?

Reply:
sure